sklearn/datasets/value_dropper.py

```diff
@@ -0,0 +1,235 @@
+# Author : Raghav RV <rvraghav93@gmail.com>
+#
+# Licence : BSD 3 clause
+
+import numpy as np
+
+from ..utils import check_X_y
+from ..utils import check_random_state
+from ..utils.multiclass import type_of_target
+
+
+__all__ = ["drop_values"]
+
+
+def drop_values(X, y=None,
+                missing_values="NaN",
+                drop_fraction=0.1,
+                label_correlation=1.0,
+                correlated_labels=1,
+                return_correlated_labels=False,
+                copy=False,
+                random_state=None):
+    """Drop values based on a preset strategy.
+
+    Attributes
+    ----------
+
+    X : ndarray like of shape (n_features, n_samples)
+        Data, in which the values must be dropped and set to
+        ``missing_values``.
+
+    y : ndarray like of shape (n_samples,), optional
+        Target, in a supervised classification task.
+
+        It can also be used for regression targets if the ``label_correlation``
+        is 0. (MCAR missingness).
+
+    missing_values : {"NaN" (or np.nan) | int}, default "NaN"
+        The value to insert to indicate missingness.
+
+    drop_fraction : float in range [0, 1], default 0.1
+        The fraction of values that should be dropped.
```

Comment:
This description appears to be outdated. As does the parameter name: now we're talking about the target proportion missing in the entire dataset.

I'll admit that with repeated runs with different correlated labels, this interpretation of the parameter might be a bit confusing: the user might thing that specifying `correlated_labels` means the `missing_fraction` only pertains to that portion of the data.

I'm not altogether sure that my suggestion of this being an overall target missing quantity was a good one.